raiden/blockchain_events_handler.py

```diff
@@ -282,18 +283,40 @@ def handle_channel_settled(raiden: 'RaidenService', event: Event):
         ),
     )
 
-    if channel_state:
-        channel_settled = ContractReceiveChannelSettled(
-            canonical_identifier=CanonicalIdentifier(
-                chain_identifier=chain_state.chain_id,
-                token_network_address=token_network_identifier,
-                channel_identifier=channel_identifier,
-            ),
-            transaction_hash=transaction_hash,
-            block_number=block_number,
-            block_hash=block_hash,
-        )
-        raiden.handle_and_track_state_change(channel_settled)
+    if not channel_state:
+        return
+
+    payment_channel: PaymentChannel = raiden.chain.payment_channel(
+        token_network_address=token_network_identifier,
+        channel_id=channel_identifier,
+    )
+    token_network: TokenNetwork = payment_channel.token_network
+    participants_details = token_network.detail_participants(
+        participant1=channel_state.our_state.address,
+        participant2=channel_state.partner_state.address,
+        block_identifier=block_hash,
+        channel_identifier=channel_identifier,
+    )
+
+    our_details = participants_details.our_details
+    our_locksroot = our_details.locksroot
+
+    partner_details = participants_details.partner_details
+    partner_locksroot = partner_details.locksroot
+
+    channel_settled = ContractReceiveChannelSettled(
+        transaction_hash=transaction_hash,
+        canonical_identifier=CanonicalIdentifier(
```

Comment:
since #3616 you can just say `canonical_identifier=channel_state.canonical_identifier`